### ISSUE TITLE
Think about splitting the 2 modes

### ISSUE THREAD

**Issue by psalaets** [psalaets/vue-jscodeshift-adapter#5]
This would be a breaking change.

Right now this module has two "modes" that are enabled at the same time:

### jscodeshift mode

Passes the sfc's `<script>` content to the transform as `fileInfo.source` so regular jscodehift-compatible codemods can be run on `.vue` files.

### vue sfc mode

Passes the sfc's `<script>`, `<template>` and `<style>` to the transform as `fileInfo.script.content`, `fileInfo.template.content` and `fileInfo.style.content` respectively so the transform do vue specific stuff.

## Proposed change

The adapter runs in one mode or the other, not both. The default is jscodeshift mode. Users can opt-in to vue sfc mode:

```js
const adapt = require('vue-jscodeshift-adapter');

const settings = {
  sfcOnly: true
}

module.exports = adapt(transform, settings);
```

`settings.sfcOnly == false` puts it in jscodeshift mode and means you're only processing the `<script>` blocks of vue files and regular js files (after #4 is done).

`settings.sfcOnly == true` puts it in sfc mode and means you're only running it on `.vue` files and manipulating the blocks through `fileInfo.script`, `fileInfo.style`, `fileInfo.template`.

**Comment by psalaets** [comment]
"vue only" mode is on hold until the api is figured out.

jscodeshift mode is the default and only mode right now
